.github/workflows/custom_image_generation.yml

```diff
@@ -0,0 +1,159 @@
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - id: 'auth'
+      name: 'Authenticate to Google Cloud'
+      uses: 'google-github-actions/auth@v0'
+      with:
+        # Centralized in dsp-tools-k8s; ask in #dsp-devops-champions for help troubleshooting
+        workload_identity_provider: 'projects/1038484894585/locations/global/workloadIdentityPools/github-wi-pool/providers/github-wi-provider'
+        service_account: 'image-build-account@broad-dsp-gcr-public.iam.gserviceaccount.com'
+
+    - name: Create svc acct file
+      run: 'echo "$GCP_DAISY_GCR_KEY" > "$GITHUB_WORKSPACE/jenkins/gce-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME"'
+      shell: bash
+      env:
+        GCP_DAISY_GCR_KEY: ${{secrets.GCP_DAISY_GCR_KEY}}
+
+    - name: run script
+      run: |
+        gsutil ls $GCE_IMAGE_BUCKET || gsutil mb -p $GOOGLE_PROJECT -l $REGION $GCE_IMAGE_BUCKET
+        docker pull $DAISY_IMAGE
+        docker run -i --rm -v $GITHUB_WORKSPACE/jenkins/gce-custom-images:/gce-custom-images \
+        $DAISY_IMAGE \
+        -project $GOOGLE_PROJECT \
+        -zone $ZONE \
+        -oauth /gce-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME \
+        -gcs_path $GCE_IMAGE_BUCKET \
+        -default_timeout 45m \
+        -var:base_image $GCE_BASE_IMAGE \
+        -var:output_image "${OUTPUT_BASE_NAME}-gce-cos-image-${GITHUB_SHA:0:6}" \
+        -var:gce_images_dir /gce-custom-images \
+        -var:installation_script_name prepare_gce_image.sh \
+        /gce-custom-images/gce_image.wf.json
+        docker stop daisy || true
+        docker rm -f daisy || true
+        
+        # Daisy doesn't clean it up all so we remove the bucket manually
+        gsutil rm -r $GCE_IMAGE_BUCKET
+        
+        # Make the image public
+        gcloud beta compute images add-iam-policy-binding \
+        projects/${GOOGLE_PROJECT}/global/images/${OUTPUT_BASE_NAME}-gce-cos-image-${GITHUB_SHA:0:6} \
+        --member='allAuthenticatedUsers' \
+        --role='roles/compute.imageUser'
+
+  run-image-generate-script-dataproc:
+    runs-on: ubuntu-latest
+    # needs: inspect-changes
+    # if: needs.inspect-changes.outputs.has-dataproc-changes == 'true' && github.event.inputs.run-dataproc-override == 'true'
+    env:
+      DATAPROC_IMAGE_VERSION: 2.0.51-debian10
+      DATAPROC_IMAGE_VERSION_FORMATTED: 2-0-51-debian10
+      DATAPROC_IMAGE_BUCKET: gs://leo-dataproc-image-creation-logs
+      TEST_NAME: "${OUTPUT_BASE_NAME}-gce-cos-image-${GITHUB_SHA:0:6}"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: test step 2
+        run: echo "test 2"
+
+      - name: Create svc acct file
+        run: |
+          echo "$GCP_DAISY_GCR_KEY" > "$GITHUB_WORKSPACE/jenkins/dataproc-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME"
+          echo 'echoing test name'
+          echo "$TEST_NAME"
+          echo 'catting'
+          cat $GITHUB_WORKSPACE/jenkins/dataproc-custom-images/$GOOGLE_CREDENTIAL_FILE_NAME
```

Comment:
Please never cat the contents of a service account file. Github actions logs are public, so any one who can see the leo repo could see this key. This should be done the same way as on line 74.

Reply:
ahh... this was just a sanity check, its not working currently, I just wanted to check if the file was indeed empty